server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/EmailSet.scala

```diff
@@ -76,25 +98,89 @@ case class EmailCreationRequest(mailboxIds: MailboxIds,
+    inReplyTo.flatMap(_.asString).map(new RawField("In-Reply-To", _)).foreach(builder.setField)
+    messageId.flatMap(_.asString).map(new RawField(FieldName.MESSAGE_ID, _)).foreach(builder.setField)
+    subject.foreach(value => builder.setSubject(value.value))
+    from.flatMap(_.asMime4JMailboxList).map(_.asJava).foreach(builder.setFrom)
+    to.flatMap(_.asMime4JMailboxList).map(_.asJava).foreach(builder.setTo)
+    cc.flatMap(_.asMime4JMailboxList).map(_.asJava).foreach(builder.setCc)
+    bcc.flatMap(_.asMime4JMailboxList).map(_.asJava).foreach(builder.setBcc)
+    sender.flatMap(_.asMime4JMailboxList).map(_.asJava).map(Fields.addressList(FieldName.SENDER, _)).foreach(builder.setField)
+    replyTo.flatMap(_.asMime4JMailboxList).map(_.asJava).foreach(builder.setReplyTo)
+    sentAt.map(_.asUTC).map(_.toInstant).map(Date.from).foreach(builder.setDate)
+
+    validateHtmlBody.flatMap(maybeHtmlBody => {
+      if (attachments.isDefined) {
+        createMultipartWithAttachments(maybeHtmlBody, attachments.get, attachmentManager, attachmentContentLoader, mailboxSession)
+          .map(multipartBuilder => {
+            builder.setBody(multipartBuilder)
+            builder.build
+          })
+      } else {
+        Right(builder.setBody(maybeHtmlBody.getOrElse(""), SubType.HTML_SUBTYPE, StandardCharsets.UTF_8).build)
+      }
+    })
+  }
+
+  private def createMultipartWithAttachments(maybeHtmlBody: Option[String],
+                                             attachments: List[Attachment],
+                                             attachmentManager: AttachmentManager,
+                                             attachmentContentLoader: AttachmentContentLoader,
+                                             mailboxSession: MailboxSession): Either[ErrorCreatingAttachmentException, MultipartBuilder] = {
+    val multipartBuilder = MultipartBuilder.create(SubType.MIXED_SUBTYPE)
+    val bodypartBuilder = BodyPartBuilder.create()
+    val inlineAttachments = attachments.filter(_.isInline)
+    val otherAttachments = attachments.filter(!_.isInline)
+
+    if (inlineAttachments.nonEmpty) {
+      val relatedMultipartBuilder: MultipartBuilder = MultipartBuilder.create(SubType.RELATED_SUBTYPE)
+      relatedMultipartBuilder.addBodyPart(bodypartBuilder.setBody(maybeHtmlBody.getOrElse(""), SubType.HTML_SUBTYPE, StandardCharsets.UTF_8))
+
+      inlineAttachments
+        .flatMap(attachment => getAttachmentMetadata(attachment, attachmentManager, mailboxSession))
+        .foreach(attachmentMetadata => relatedMultipartBuilder.addBodyPart(
+          bodypartBuilder.setBody(attachmentContentLoader.load(attachmentMetadata, mailboxSession).readAllBytes(), attachmentMetadata.getType.asString())
+            .setContentDisposition("inline")))
+
+      multipartBuilder
+        .addBodyPart(BodyPartBuilder.create.setBody(maybeHtmlBody.getOrElse(""), SubType.PLAIN_SUBTYPE, StandardCharsets.UTF_8))
```

Comment:
Can you explain me why the html body is set up twice, one time on relatedMultipartBuilder, one time on multipartBuilder?

(we will need tests mixing attachments & textHtml BTW)